### PR TITLE
Update existing cluster with Azure Stack recommended configuration.

### PR DESCRIPTION
**Pull request by rjaini**
Update existing cluster with Azure Stack recommended configuration.

@jadarsie : Check-in file in diagnosis folder so that we can use it for TSG as well.

### CODE REVIEW

File: diagnosis/scripts/updateconfiguration.sh [hunk 1]
```diff
@@ -0,0 +1,162 @@
+#!/bin/bash -e
+
```
Comment (jadarsie) [inline diagnosis/scripts/updateconfiguration.sh]: remove extra lines please

File: diagnosis/scripts/updateconfiguration.sh [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+# kubelet
+sudo cp /etc/default/kubelet /etc/default/kubelet.bak
+sudo sed -i -e 's/--node-status-update-frequency=[0-9]*[a-z]/--node-status-update-frequency=1m/' /etc/default/kubelet
+
+# restart
+sudo systemctl daemon-reload
+sudo systemctl restart kubelet
+
+echo "=> azure.json updates"
+sudo grep -E 'cloudProviderRatelimit"|cloudProviderRateLimitQPS"|cloudProviderRateLimitBucket"' /etc/kubernetes/azure.json
```
Comment (jadarsie) [inline diagnosis/scripts/updateconfiguration.sh]: this line is not showing all the changes made to azure.json

File: diagnosis/scripts/updateconfiguration.sh [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+echo "=> kubelet restarted"
+systemctl status kubelet --no-pager -l
```
Comment (jadarsie) [inline diagnosis/scripts/updateconfiguration.sh]: on large clusters, a kubelet restart shows iptables exiting with status code 4
to make this script solid, we should retry if that happens

File: diagnosis/scripts/updateconfiguration.sh [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+sudo cp /etc/kubernetes/manifests/kube-controller-manager.yaml /etc/kubernetes/manifests/kube-controller-manager.bak
+sudo sed -i -e 's/--route-reconciliation-period=[0-9]*[a-z]/route-reconciliation-period=1m/' /etc/kubernetes/manifests/kube-controller-manager.yaml
+sudo sed -i -e 's/--node-monitor-grace-period=[0-9]*[a-z]/--node-monitor-grace-period=5m/' /etc/kubernetes/manifests/kube-controller-manager.yaml
+sudo sed -i -e 's/--pod-eviction-timeout=[0-9]*[a-z]/--pod-eviction-timeout=5m/' /etc/kubernetes/manifests/kube-controller-manager.yaml
+
+# restart
+sudo systemctl daemon-reload
+sudo systemctl restart kubelet
+
+echo "=> controller-manager updates"
```
Comment (jadarsie) [inline diagnosis/scripts/updateconfiguration.sh]: not logging pod-eviction-timeout

File: diagnosis/scripts/updateconfiguration.sh [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    echo ""
+    echo "Examples:"
+    echo "  $0 --agents"
+    echo "  $0 --masters"
+    echo "  $0 --masters --agents"
+    exit 1
+}
+
+
+if [ $# -eq 0 ]; then
```
Comment (jadarsie) [inline diagnosis/scripts/updateconfiguration.sh]: remove please

File: diagnosis/scripts/updateconfiguration.sh (hunk 2 above)
Comment (rjaini) [reply diagnosis/scripts/updateconfiguration.sh]: Updated.

File: diagnosis/scripts/updateconfiguration.sh (hunk 1 above)
Comment (rjaini) [reply diagnosis/scripts/updateconfiguration.sh]: removed. the spaces.

File: diagnosis/scripts/updateconfiguration.sh (hunk 3 above)
Comment (rjaini) [reply diagnosis/scripts/updateconfiguration.sh]: For this I will open an issue. Which can go in later checkin.

File: diagnosis/scripts/updateconfiguration.sh (hunk 4 above)
Comment (rjaini) [reply diagnosis/scripts/updateconfiguration.sh]: Updated.

File: diagnosis/scripts/updateconfiguration.sh (hunk 5 above)
Comment (rjaini) [reply diagnosis/scripts/updateconfiguration.sh]: removed.

File: diagnosis/scripts/updateconfiguration.sh (hunk 3 above)
Comment (rjaini) [reply diagnosis/scripts/updateconfiguration.sh]: We are tracking this issue as https://github.com/msazurestackworkloads/azurestack-gallery/issues/198

**Pull request merged**